### PR TITLE
New templates

### PR DESCRIPTION
**Pull request by coolo**
This is the beginning of the "improved" job templates page:
https://progress.opensuse.org/issues/6476

The first step is to get the database schema updated to reflect that priorities are per job template - and adding a job group model. the job templates are now  a plain table ... and are horrible.

So I didn't (re-)add any update or create functionality as this table is just *too* unusable, so we need to discuss this first before I add an upgrade - and adapt the test cases to it. They are disabled for now

screencapture-localhost-9526-admin-job_templates

### CODE REVIEW

**Comment by lnussel** [comment]
maybe we need different views on the same data, depending on what one wants to change

**Comment by coolo** [comment]
that raises the question of user stories indeed.

Let's collect:
 * Richard adding his new kde5 test suite adding to all existant media types
 * Stephan removing all "nice" test suites from !promo media
 * Dinar adding a new media type and adding tons of tests for one machine
 * Marita changing all sles tests's job group

**Comment by coolo** [comment]
Another approach to discuss:

screencapture-2

**Comment by coolo** [comment]
for convenience in discussing I'm uploading current state:
screencapture-3

**Comment by ancorgs** [comment]
Something to add to the mock attached by @coolo, It will not show all the possible job templates, but only the ones for a given job group (new concept to implement what a product manager would call "product").

So the user would go to: Job groups -> SLE12 and there she/he would find that screen. Initially empty and then he/she would start adding those boxes, one per media type (what we used to call "products").

**Comment by marita22** [comment]
"Something to add" could be 2 rows of Tabs:
first row:   openSUSE*DVD, openSUSE-Net, openSUSE-Gome, ...
second row as subtabs of one selected tab: x86_64, i586, ...


Or for the SLE family:
first row: SLES, SLED, HA
second row: arches

**Comment by coolo** [comment]
this is the HTML version of our paper drawings

screencapture-4

**Comment by coolo** [comment]
without the boxes:

screencapture-5

**Comment by coolo** [comment]
met with the designer

screencapture-6

**Comment by coolo** [comment]
does someone fancy reviewing all this or should I just merge? :)

**Comment by ancorgs** [comment]
I plan to review it today at some point

**Comment by coolo** [comment]
ok, no hurry - just wanting to know the plans

File: lib/OpenQA/Schema/Result/Products.pm [hunk 1]
```diff
@@ -58,4 +58,13 @@ sub name {
     join('-', map { $self->$_ } qw/distri version flavor arch/);
 }
 
+sub mediagroup {
```
Comment (ancorgs) [inline lib/OpenQA/Schema/Result/Products.pm]: A comment about the meaning and purpose would be highly appreciated.

**Comment by ancorgs** [comment]
Too much code to review on a Monday, but in general looks good.

File: lib/OpenQA.pm [hunk 2]
```diff
@@ -300,25 +300,16 @@ sub startup {
...
 
-    $admin_r->get('/job_templates')->name('admin_job_templates')->to('job_template#index');
-    $admin_r->post('/job_templates')->to('job_template#update');
+    $admin_r->get('/job_templates/:groupid')->name('admin_job_templates')->to('job_template#index');
+
+
+    $admin_r->get('/groups')->name('admin_groups')->to('job_group#index');
+    $admin_r->post('/groups')->name('admin_new_group')->to('job_group#create');
+    $admin_r->get('/groups/connect/:groupid')->name('job_group_new_media')->to('job_group#connect');
+    $admin_r->post('/groups/connect/:groupid')->name('job_group_save_media')->to('job_group#save_connect');
```
Comment (aaannz) [inline lib/OpenQA.pm]: We already have API to manage suites, etc. Shouldn't eg. connect and save_connect be made API calls?

Comment (coolo) [inline lib/OpenQA.pm]: feel free

**Comment by coolo** [comment]
rebased

**Comment by coolo** [comment]
now added another migration script to set the group in jobs - seems to work fine: 8b2e59d

File: dbicdh/_common/upgrade/26-27/005-pick-groups.pl [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        my $group = $r->distri;
+        if ($r->distri eq 'sle' && $r->flavor =~ m/^Server/) {
+            $group = 'sles';
+        }
+        if ($r->distri eq 'sle' && $r->flavor =~ m/^Desktop/) {
+            $group = 'sled';
+        }
+        if ($r->version ne '*') {
+            $group .= "-" . $r->version;
+        }
```
Comment (coolo) [inline dbicdh/_common/upgrade/26-27/005-pick-groups.pl]: what I wonder: perhaps we should stop with "openSUSE-13.2" and use space as seperator?

**Comment by coolo** [comment]
Learning bootstrap:

screencapture-dashboard

**Comment by coolo** [comment]
merged manually

**Pull request closed without merging**